pennylane/measurements/shots.py

```diff
@@ -0,0 +1,123 @@
+# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This module contains the Shots class to hold shot-related information."""
+# pylint:disable=inconsistent-return-statements
+from collections import namedtuple
+from typing import Sequence, Tuple
+
+ShotCopies = namedtuple("ShotCopies", ["shots", "copies"])
+"""A namedtuple that represents a shot quantity being repeated some number of times."""
+
+
+def valid_int(s):
+    """Returns True if s is a positive integer."""
+    return isinstance(s, int) and s > 0
+
+
+def valid_tuple(s):
+    """Returns True if s is a tuple of the form (shots, copies)."""
+    return isinstance(s, tuple) and len(s) == 2 and valid_int(s[0]) and valid_int(s[1])
+
+
+class Shots:
+    """
+    A data class that stores shot information.
+
+    Args:
+        shots (Union[None, int, Tuple[int, int], Sequence[int, Tuple[int, int]]]): Raw shot information
+
+    Defining shots enables users to specify circuit executions, and the Shots class standardizes
+    the internal representation of shots. There are three ways to specify shot values:
+
+    * A positive integer
+    * A tuple consisting of a pair of positive integers of the form ``(shots, copies)``
+    * A list of multiple shot values matching one or both of the above value types
+
+    The tuple-pair of the form ``(shots, copies)`` is represented internally by a namedtuple called
+    ``ShotCopies``. The first value is the number of shots to execute, and the second value is the
```

Comment:
```suggestion
    :class:`~ShotCopies`. The first value is the number of shots to execute, and the second value is the
```

Reply:
I didn't do this because it's barely a class (it's a namedtuple), but I can if you'd prefer it. It's [not the most beautiful page](https://xanaduai-pennylane--3682.com.readthedocs.build/en/3682/code/api/pennylane.measurements.ShotCopies.html)